core/global.ts

```diff
@@ -0,0 +1,9 @@
+import { routing } from '~/i18n/routing';
+import messages from '~/messages/en.json';
+
+declare module 'next-intl' {
+  interface AppConfig {
+    Locale: (typeof routing.locales)[number];
```

Comment:
Note, `Locale` is still currently type as `string` because we retrieve locales from the build config after dynamically retrieving them from the GraphQL API. If instead we generated a JSON file that was committed to version control, then we could have strongly typed locales. I decided to leave that out of scope for this PR but it's a good exploration for future work. More details for strongly typed locales in the docs: https://next-intl.dev/docs/workflows/typescript#locale